scripts/lib/api/generateApiComponents.ts

```diff
@@ -87,17 +87,42 @@ function prepareProps(
   id: string,
 ): ComponentProps {
   const preparePropsPerApiType: Record<string, () => ComponentProps> = {
-    class: () => prepareClassProps($child, $dl, githubSourceLink, id),
+    class: () =>
+      prepareClassOrExceptionProps($, $main, $child, $dl, githubSourceLink, id),
     property: () =>
-      prepareAttributeProps($, $child, $dl, priorApiType, githubSourceLink, id),
+      prepareAttributeProps(
+        $,
+        $main,
+        $child,
+        $dl,
+        priorApiType,
+        githubSourceLink,
+        id,
+      ),
     method: () =>
```

Comment:
I was thinking that, probably, we can merge `prepareMethodProps` and `prepareFunctionProps`, but I think it's better to explore this in a follow-up that will have as a goal to simplify this script.

Reply:
That's a cool idea! As discussed in our 1:1, I'll implement this and try to simplify the script in a follow-up :+1: